### PR TITLE
 Contact Journal Extension - Show RAT and PCR Tests results in Contact Journal (EXPOSUREAPP-6040)

### PR DESCRIPTION
**Pull request by jurajkusnier**
image

### Testing
- Scan PCR/RAT test
- See test results
- If the test is positive or negative new record should be added to DB
- Check Contact Journal to see the corona test record

### CODE REVIEW

**Review by ChristinaTsigganos — DISMISSED** [review]
UA approved

File: Corona-Warn-App/src/main/res/layout/contact_diary_overview_day_list_item_test_results.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <View
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/card_divider"
+        android:background="?android:attr/listDivider"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/contact_diary_overview_day_list_item_test_results.xml:11]: is this by design? can we use the same color that we use already ? I think it is called `hairLineColor`

File: Corona-Warn-App/src/main/res/layout/contact_diary_overview_day_list_item_test_results.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        app:layout_constraintEnd_toEndOf="parent" />
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recycler_view"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:nestedScrollingEnabled="false"
+        android:paddingVertical="@dimen/spacing_tiny"
+        tools:itemCount="1"
+        tools:listitem="@layout/contact_diary_overview_day_list_item_test_result"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/contact_diary_overview_day_list_item_test_results.xml:23]: Please format the file

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/contactdiary/ui/overview/ContactDiaryOverviewViewModelTest.kt [hunk 3]
```diff
@@ -69,6 +69,7 @@ open class ContactDiaryOverviewViewModelTest {
         every { taskController.submit(any()) } just runs
         every { contactDiaryRepository.locationVisits } returns flowOf(emptyList())
         every { contactDiaryRepository.personEncounters } returns flowOf(emptyList())
+        every { contactDiaryRepository.testResults } returns flowOf(emptyList())
```
Comment (mtwalli) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/contactdiary/ui/overview/ContactDiaryOverviewViewModelTest.kt:72]: should we consider Screenshots tests here or we have another task for that ?

File: Corona-Warn-App/src/main/res/layout/contact_diary_overview_day_list_item_contact.xml [hunk 4]
```diff
@@ -18,6 +19,8 @@
         android:layout_height="wrap_content"
         android:nestedScrollingEnabled="false"
         android:paddingVertical="@dimen/spacing_tiny"
+        tools:itemCount="2"
+        tools:listitem="@layout/contact_diary_overview_nested_list_item"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/contact_diary_overview_day_list_item_contact.xml:23]: please format the file ?

File: Corona-Warn-App/src/main/res/layout/contact_diary_overview_day_list_item_test_results.xml (hunk 1 above)
Comment (jurajkusnier) [reply Corona-Warn-App/src/main/res/layout/contact_diary_overview_day_list_item_test_results.xml:11]: I'm using the same colors and styles as the rest of the contact diary day overview components such as risk events or contacts to keep it consistent

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/contactdiary/ui/overview/ContactDiaryOverviewViewModelTest.kt (hunk 3 above)
Comment (jurajkusnier) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/contactdiary/ui/overview/ContactDiaryOverviewViewModelTest.kt:72]: there is a separated subtask for this so it will be in another PR

**Comment by jurajkusnier** [comment]
> When the test is pending and then changed to Negative , it is not appearing in the journal?
> Scenario:
> 1- Scan a pending RAT test
> 2- Change it Negative
> 3- it is not showing in the journal

Should be there. If not, than it's a bug.

**Comment by mtwalli** [comment]
Tested on Samsung S8:
(PCR , RAT )Negative, Positive and Pending that turns into POS or NEG
All added to the journal
Screenshot 2021-05-26 at 11 51 50

**Review by mtwalli — DISMISSED** [review]
👍🏼

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/entity/ContactDiaryCoronaTestEntity.kt [hunk 5]
```diff
@@ -38,7 +38,7 @@ data class ContactDiaryCoronaTestEntity(
 }
 
 fun CoronaTest.canBeAddedToJournal(): Boolean {
-    return isViewed && (isNegative || isPositive)
+    return isNegative || (isViewed && isPositive)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/entity/ContactDiaryCoronaTestEntity.kt:41]: Can you please add  a comment here why did you change it ?

Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/entity/ContactDiaryCoronaTestEntity.kt]: sure

**Review by mtwalli — APPROVED** [review]

**Review by BMItr — APPROVED** [review]
nothing to complain 👌

**Review by NataliaLemmerth — COMMENTED** [review]
UA approved
